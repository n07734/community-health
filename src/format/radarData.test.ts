import formatRadarData from './radarData'

describe('formatRadarData:', () => {
    const usersData:any[] = [
        {
            age: 4,
            approvalsGiven: 2,
            approvalsReceived: 2,
            codeCommentsGiven: 2,
            codeCommentsReceived: 2,
            commentsGiven: 6,
            commentsReceived: 2,
            generalCommentsGiven: 2,
            generalCommentsReceived: 2,
            prSize: 400,
            totalPRs: 2,
            uniquePRsContributedTo: 4,
            uniquePRsApproved: 4,
            user: 'bob',
            author: 'bob',
        },
        {
            age: 2,
            approvalsGiven: 2,
            approvalsReceived: 2,
            codeCommentsGiven: 2,
            codeCommentsReceived: 2,
            commentsGiven: 2,
            commentsReceived: 6,
            generalCommentsGiven: 2,
            generalCommentsReceived: 2,
            prSize: 200,
            totalPRs: 0,
            uniquePRsContributedTo: 0,
            uniquePRsApproved: 0,
            user: 'BBB',
        },
    ]

    const base = formatRadarData(usersData)

    it('averagedData', () => {
        expect(base.averagedData).toMatchSnapshot()
    })

    it('maxValues', () => {
        expect(base.maxValues).toMatchSnapshot()
    })

    it('filter author', () => {
        const {
            averagedData,
            maxValues,
            user,
        } = formatRadarData(usersData, 'bob') as any

        expect(user.user).toEqual('bob')
        expect(maxValues).toEqual(base.maxValues)
        expect(averagedData).toMatchSnapshot()
    })
})
