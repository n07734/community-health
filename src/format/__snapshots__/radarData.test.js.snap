// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatRadarData: averagedData 1`] = `
{
  'age': 3,
  'approvalsGiven': 2,
  'approvalsReceived': 2,
  'codeCommentsGiven': 2,
  'codeCommentsReceived': 2,
  'commentsGiven': 4,
  'commentsReceived': 4,
  'generalCommentsGiven': 2,
  'generalCommentsReceived': 2,
  'prSize': 300,
  'totalPRs': 1,
  'uniquePRsApproved': 2,
  'uniquePRsContributedTo': 2,
  'user': 'Peers',
}
`;

exports[`formatRadarData: filter author 1`] = `
{
  'age': 2,
  'approvalsGiven': 2,
  'approvalsReceived': 2,
  'codeCommentsGiven': 2,
  'codeCommentsReceived': 2,
  'commentsGiven': 2,
  'commentsReceived': 6,
  'generalCommentsGiven': 2,
  'generalCommentsReceived': 2,
  'prSize': 200,
  'totalPRs': 0,
  'uniquePRsApproved': 0,
  'uniquePRsContributedTo': 0,
  'user': 'Peers',
}
`;

exports[`formatRadarData: maxValues 1`] = `
{
  'age': 4,
  'approvalsGiven': 2,
  'approvalsReceived': 2,
  'codeCommentsGiven': 2,
  'codeCommentsReceived': 2,
  'commentsGiven': 6,
  'commentsReceived': 6,
  'generalCommentsGiven': 2,
  'generalCommentsReceived': 2,
  'prSize': 400,
  'totalPRs': 2,
  'uniquePRsApproved': 4,
  'uniquePRsContributedTo': 4,
}
`;
